include/boost/process/pipe.hpp

```diff
@@ -265,31 +265,40 @@ class basic_ipstream : public std::basic_istream<CharT, Traits>
     basic_pipebuf<CharT, Traits>* rdbuf() {return &_buf;};
 
     ///Default constructor.
-    basic_ipstream() : std::basic_istream<CharT, Traits>(nullptr)
+    basic_ipstream() : std::basic_istream<CharT, Traits>(&_buf)
     {
```

Comment:
That is undefined behavior, because you're passing a pointer to an unitialized object.